src/js_native_api_v8.cc

```diff
@@ -1651,6 +1651,18 @@ node_api_create_external_string_utf16(napi_env env,
       });
 }
 
+napi_status NAPI_CDECL node_api_create_property_key_utf16(napi_env env,
+                                                          const char16_t* str,
+                                                          size_t length,
+                                                          napi_value* result) {
+  return v8impl::NewString(env, str, length, result, [&](v8::Isolate* isolate) {
+    return v8::String::NewFromTwoByte(isolate,
+                                      reinterpret_cast<const uint16_t*>(str),
+                                      v8::NewStringType::kInternalized,
+                                      static_cast<int>(length));
```

Comment:
Why do we need to cast to `int` here? We don't cast to `int` in the [regular version](https://github.com/nodejs/node/blob/ba957a61f83365d9df7126c23a1c78c320061414/src/js_native_api_v8.cc#L1591-L1594) of this function.

Reply:
greetings I made this place with the suggestion of @vmoroz with reference to this pr #48339

Reply:
@gabrielschulhof , we are getting warnings in MSVC compiler if types are mismatch.
The `NewFromTwoByte` expects type `int` which is signed 32-bit integer.
In this function `length` is `size_t` which is unsigned 64-bit integer for 64 platforms.
MSVC warns about possible precision loss when the cast implicitly here.
The explicit cast suppresses the warning as we explicitly express our intent.

Ideally, we must add the cast in other places too to avoid the MSVC warnings.
But it should be another PR since currently Node.js has a lot of MSVC warnings anyway.